test/date.js

```diff
@@ -153,15 +153,25 @@ describe('Date Component', () => {
           'date-field': error
         };
         date.hooks['pre-render'](req, res, next);
-        expect(res.render).to.have.been.calledWith(path.resolve(__dirname, '../templates/date.html'), { error, key: 'date-field', legend: null, hint: null });
+        expect(res.render).to.have.been.calledWith(path.resolve(__dirname, '../templates/date.html'), { error, key: 'date-field', legend: null, legendClassName: '', hint: null });
       });
 
-      it('calls res.render with template, key, legend and hint with values NOT null', () => {
+      it('calls res.render with template, key, legend and hint with values NOT null OR empy string', () => {
         req.translate = sinon.stub();
         req.translate.withArgs('fields.date-field.legend').returns('you legend');
         req.translate.withArgs('fields.date-field.hint').returns('some hint');
         date.hooks['pre-render'](req, res, next);
-        expect(res.render).to.have.been.calledWith(path.resolve(__dirname, '../templates/date.html'), { error: undefined, key: 'date-field', legend: 'you legend', hint: 'some hint' });
+        expect(res.render).to.have.been.calledWith(path.resolve(__dirname, '../templates/date.html'), { error: undefined, key: 'date-field', legendClassName: '', legend: 'you legend', hint: 'some hint' });
```

Comment:
I was trying to work this out.  Now I see the value of this.  You can use `sinon.match({})` so that you are only testing specific properties rather than checking for everything which isn't exactly relevant to the test